### ISSUE TITLE
Transitive dependency modules need to be explicitly imported in the app

### ISSUE THREAD

**Issue by agu-z** [roc-lang/roc#5477]
[This app](https://github.com/agu-z/roc-pg/blob/71a7d111edabdf45ad28e24c04d617a8585715bb/examples/query.roc#L1) depends on two packages: `pg` and `sql`.  

The `sql` package also depends on  `Pg.Cmd` and `Pg.Result` from `pg`, so its [`main.roc`](https://github.com/agu-z/roc-pg/blob/71a7d111edabdf45ad28e24c04d617a8585715bb/sql/src/main.roc#L4)'s `package` header references it.

The app depends on `Pg.Client` directly, but it only depends on `Pg.Cmd` and `Pg.Result` transitively through `Sql`, so it shouldn't require imports to these.

However, if we don't import them in the app, we will get a compiler error like the following for every reference on `Sql`: 

```
── MODULE NOT IMPORTED ─────────────────────────── sql/examples/../src/Sql.roc ─

The `Pg.Cmd` module is not imported:

240│      { options & limit: [Raw " limit ", Param (Pg.Cmd.nat max)] }
                                                    ^^^^^^^^^^

Did you mean to import it?
```

The `Sql` module [includes](https://github.com/agu-z/roc-pg/blob/71a7d111edabdf45ad28e24c04d617a8585715bb/sql/src/Sql.roc#L54-L55) imports to `Pg.Cmd` and `Pg.Result`, so those references should be ok.

**Comment by agu-z** [comment]
I think this is caused by the compiler sharing `ModuleIds` across packages

**Comment by agu-z** [comment]
Fixed by #6658
